soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -440,6 +441,116 @@ void Randomizer::LoadMerchantMessages(const char* spoilerFileName) {
+                    "!%w&Ich kann versichern es ist ein&aufrichtiges Angebot! Ich hoffe es&lässt dich nicht %bkalt%w!",
+                    "!%w&C'est vrai! J'te jure!&Je l'avais laissé %bau frais exprès%w pour toi!"
+                };
+            } else {
+                cgItemName = EnumToSpoilerfileGetName[carpetGuyGet];
+                cgBoxTwoText = {
+                    "!%wIt's real, I promise!&A lonely man such as myself&wouldn't %rlie%w to you, hmm?^",
+                    "!%w&Ich kann versichern es ist ein&aufrichtiges Angebot!^Ein einsamer Mann wie ich würde dich&doch nicht %ranlügen%w, oder?^",
+                    "!%w&C'est vrai! J'te jure!&Un gars comme moi ne te %rmentirai%w pas&tu ne crois pas?^"
+                };
+            }
+        } else {
+            cgItemName = {
+                "A mystery",
+                "Ein Geheimnis",
+                "Un secret"
+            };
+            cgBoxTwoText = {
+                "!%w&I won't tell you what it is until I see&the money...^",
+                "!%w&Erst kommt das Geld, dann die Ware...^",
+                "!%w&Je ne te dirai pas ce que c'est avant&d'être payé rubis sur l'ongle...^"
+            };
+        }
+    } else {
+        cgItemName = {
+            "A dangerous, running object",
+            "Ein bombiges, ferngesteuertes&Objeckt", //Pulled from original game text
+            "Une arme qu'elle est tirrible" //Pulled from original game text
+        };
+        cgBoxTwoText = {
+            "!%w&Terrifying!&I won't tell you what it is until&I see the money...^",
+            "!%w&Erst kommt das Geld, dann die Ware...^",
+            "!%w&Je ne te dirai pas ce que c'est avant&d'être payé rubis sur l'ongle...^"
+        };
+    }
+    CustomMessageManager::Instance->CreateMessage(
+        Randomizer::merchantMessageTableID, TEXT_CARPET_SALESMAN_1,
+        {
+            TEXTBOX_TYPE_BLACK,
+            TEXTBOX_POS_BOTTOM,
+            "Welcome!^I am selling stuff, strange and rare, &from all over the world to everybody.&Today's special is...^%r" + 
+            cgItemName[0] + cgBoxTwoText[0] + "How about %g200 Rupees?%w\x1B&&%gYes&No%w",
+            "Sei gegrüßt!^Ich verkaufe allerlei Kuriorisäten.&Stets sonderliche und seltene Ware&aus aller Welt für jedermann.&Das heutige Angebot bleibt...^%r" + 
+            cgItemName[1] + cgBoxTwoText[1] + "Wie wäre es mit %g200 Rubinen?%w\x1B&&%gJa!&Nein!%w",
+            "Bienvenue!^Je vends des trucs étranges et rares,&de partout dans le monde et à tout le&monde! L'objet du jour est...^%r" + 
+            cgItemName[2] + cgBoxTwoText[2] + "Alors, marché conclu pour %g200 rubis?%w\x1B&&%gOui&Non%w"
+        }
+    );
+
+    CustomMessageManager::Instance->CreateMessage(
+        Randomizer::merchantMessageTableID, TEXT_CARPET_SALESMAN_2,
+        {
+            TEXTBOX_TYPE_BLACK,
+            TEXTBOX_POS_TOP,
+            "Finally! Now I can go back to being &an %rarms dealer!%w",
```

Comment:
What does the original TEXT_CARPET_SALESMAN_2 text say? Do we need to replace this one?

Reply:
The original second textbox reveals the item (instead of "dangerous running object" it was "Bombchu", then guides you on how to reach the Spirit Temple. Similar text is shown if you decline to purchase his item.